clustering/gke.py

```diff
@@ -0,0 +1,366 @@
+    servicesIpv4Cidr: 10.75.240.0/20
+    status: RUNNING
+    zone: us-central1-f
+'''
+
+USER_AGENT = "ansible-gke-module/0.0.1"
+from httplib2 import Http
+try:
+    from oauth2client.client import SignedJwtAssertionCredentials
+    from apiclient.discovery import build
+    from googleapiclient.http import set_user_agent
+    from googleapiclient.errors import HttpError
+    HAS_GOOGLE_API_CLIENT = True
+except ImportError:
+    HAS_GOOGLE_API_CLIENT = False
+
+
+def _make_cluster_body(module):
+    cluster = {}
+    cluster['name'] = module.params.get('name')
+    if module.params.get('description'):
+        cluster['description'] = module.params.get('description')
+    cluster['initialNodeCount'] = module.params.get('node_count')
+    cluster['nodeConfig'] = {
+        'machineType': module.params.get('node_config_machine_type'),
+        'diskSizeGb': module.params.get('node_config_disksize'),
+        'oauthScopes': module.params.get('node_config_scopes'),
+    }
+    if module.params.get('node_config_metadata'):
+        cluster['nodeConfig']['metadata'] = \
+            module.params.get('node_config_metadata')
+    cluster['masterAuth'] = {
+        'username': module.params.get('username'),
+        'password': module.params.get('password')
+    }
+    if module.params.get('logging_service'):
+        cluster['loggingService'] = module.params.get('logging_service')
+    if module.params.get('monitoring_svc'):
+        cluster['monitoringService'] = module.params.get('monitoring_svc')
+    if module.params.get('network_name'):
+        cluster['network'] = module.params.get('network_name')
+    if module.params.get('ipv4_range'):
+        cluster['clusterIpv4Cidr'] = module.params.get('ipv4_range')
+    if module.params.get('zone'):
+        cluster['zone'] = module.params.get('zone')
+    body = {'cluster': cluster}
+    return body
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            credential_file=dict(required=True),
+            project_id=dict(required=True),
+
+            name=dict(required=True),
+            description=dict(required=False,
+                             default="GKE Cluster created with Ansible"),
+            node_count=dict(required=False, default=2),
+            node_config_machine_type=dict(required=False,
+                                          default="n1-standard-1"),
+            node_config_disksize=dict(required=False, default=100),
+            node_config_scopes=dict(required=False, default=[
+                "https://www.googleapis.com/auth/compute",
+                "https://www.googleapis.com/auth/devstorage.read_only",
+                "https://www.googleapis.com/auth/logging.write",
+                "https://www.googleapis.com/auth/monitoring"
+                ]
+            ),
+            node_config_metadata=dict(required=False),
+            username=dict(required=False, default="admin"),
+            password=dict(required=True),
```

Comment:
The password param should be marked as `no_log=True` to prevent leaking it during logging.